### DISCUSSION TITLE
Feature Request: LTTB downsampling

### DISCUSSION THREAD

**Discussion by wuyuanyi135** [duckdb/duckdb#14044] [Ideas]
I have been using duckdb as a timeseries data storage for a real-time analytical app. Due to the large number of data, the chart has to be downsampled before plot. In timescaledb this feature was [built-in](https://docs.timescale.com/api/latest/hyperfunctions/downsampling/). Among them i found LTTB satisfactory as it preserve most visual traits. I wish duckdb could also ship the downsampling features as plugins, which will be helpful for applications involving real-time streaming data and timeseries plotting

**Comment by carlopi** [comment]
This is not supported out-of-the-box, but would make for a great [community] extension.

**Reply by wuyuanyi135** [reply]
Agree. I would take some time to see the extension development when I got some time.
